connect/api/src/main/java/org/apache/kafka/connect/storage/ConverterConfig.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.storage;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+
+import java.util.Map;
+
+import static org.apache.kafka.common.config.ConfigDef.ValidString.in;
+
+/**
+ * Abstract class that defines the configuration options for {@link Converter} and {@link HeaderConverter} instances.
+ */
+public abstract class ConverterConfig extends AbstractConfig {
+
+    public static final String TYPE_CONFIG = "converter.type";
+    public static final String TYPE_DEFAULT = ConverterType.HEADER.getName();
```

Comment:
this seems a weird default given that headers are new? wouldn't the default be empty/nothing since currently converters don't assume default types?

Reply:
Ok, I'm fine with having null for the default. The reason I went this way was because of the non-standard `Converter.configure(Map<String, ?> configs, boolean isKey)` method that is currently used for all key and value converters, so right now the only thing that uses the `Configurable.configure(Map<String, ?> configs)` method is the header converter. However, that already sets the converter type, so I'm fine with not having a default.

Reply:
Actually, on second thought, there's no reason to have a default. As mentioned above, the current code always sets it for key, value, and header converters, so having a default is actually incorrect.